src/stores/store.jsx

```diff
@@ -2581,29 +2581,16 @@ class Store {
     }
 
     try {
-      const vaultContract = new web3.eth.Contract(asset.vaultContractABI, asset.vaultContractAddress)
-      const controllerAddress = await vaultContract.methods.controller().call({ from: account.address })
-      const controllerContract = new web3.eth.Contract(config.vaultControllerABI, controllerAddress)
+      const url = config.statsProvider + 'vaults'
+      const vaultsApiResultString = await rp(url);
+      const vaults = JSON.parse(vaultsApiResultString)
+      const vault = vaults.find((vault) => vault.address.toLowerCase() === asset.vaultContractAddress.toLowerCase())
```

Comment:
This is a much cleaner solution than what we're doing today. Looks solid. Tested locally.

Reply:
I am very happy to remove some of that special logic for various edge cases 🥳